### PR TITLE
Improve LDAP Plugin for Filedaemon

### PR DESCRIPTION
**Pull request by arogge**
This PR adds a systemtest for the Filedaemon's LDAP-Plugin and fixes a few bugs:
* you can now easily restore to a custom location using where/regexwhere
* support for objects >64k
* detect data that will not be restorable on backup (rather than reporting at restore-time)

### CODE REVIEW

File: systemtests/tests/py2plug-fd-ldap/etc/bareos/bareos-dir.d/fileset/PluginTest.conf.in [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FileSet {
+  Name = "PluginTest"
+  Description = "Test the Plugin functionality with a Python Plugin."
+  Include {
+    Options {
+      signature = MD5
+    }
+    Plugin = "@python_module_name@:module_path=@python_plugin_module_src_test_dir@:module_name=bareos-fd-ldap:uri=ldap\\://@SYSTEMTEST_LDAP_ADDRESS@:basedn=ou=backup,@SYSTEMTEST_LDAP_BASEDN@:bind_dn=@SYSTEMTEST_LDAP_BINDDN@:password=@SYSTEMTEST_LDAP_PASSWORD@"
```
Comment (pstorz) [inline systemtests/tests/py2plug-fd-ldap/etc/bareos/bareos-dir.d/fileset/PluginTest.conf.in:8]: How about using the newly available multi-line strings here?

**Review by pstorz — COMMENTED** [review]
Very impressive. Please check my comments.

File: core/src/plugins/filed/python/ldap/BareosFdPluginLDAP.py [hunk 2]
```diff
@@ -79,107 +79,116 @@ def start_backup_file(self, savepkt):
...
         """
         bareosfd.DebugMessage(
-            100,
-            "BareosFdPluginVMware:create_file() called with %s\n" % (restorepkt),
+            100, "BareosFdPluginLDAP:create_file() called with %s\n" % (restorepkt)
         )
-        if restorepkt.type == bFileType["FT_DIREND"]:
-            restorepkt.create_status = bCFs["CF_SKIP"]
-        elif restorepkt.type == bFileType["FT_REG"]:
+        if restorepkt.type == bareosfd.bFileType["FT_DIREND"]:
```
[the remark this answers is not in the document]
Comment (arogge) [reply core/src/plugins/filed/python/ldap/BareosFdPluginLDAP.py:135]: If the patches need to be backported to 19 and 18 all these changes would have to be undone during backporting. Thus I decided to keep it as backwards-compatible as possible.

File: systemtests/tests/py2plug-fd-ldap/etc/bareos/bareos-dir.d/fileset/PluginTest.conf.in (hunk 1 above)
Comment (arogge) [reply systemtests/tests/py2plug-fd-ldap/etc/bareos/bareos-dir.d/fileset/PluginTest.conf.in:8]: Same as above. Great idea, but we won't be able to backport the test to Bareos 19 as easily if we use a modern Bareos 20 feature here.

File: core/src/plugins/filed/python/ldap/BareosFdPluginLDAP.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (pstorz) [reply core/src/plugins/filed/python/ldap/BareosFdPluginLDAP.py:135]: Ok I see. That makes sense.

**Comment by pstorz** [comment]
I am happy with the current status and could run the tests locally with python-ldap 2.5.2 and 3.3.1

Good work!

**Pull request merged**